Fix #40: Add sorting and filtering to Inventory List

AKSHAY-RSOL opened the pull request:
Closes #40
## Changes
- Added filtering by Category, Condition, and Lent Status
- Added sorting by Name and Year of Purchase
- Updated UI to include filter/sort controls matching existing design
- Implemented client-side filtering logic

AKSHAY-RSOL committed:
Fix #40: Add sorting and filtering to Inventory List

- Added filter controls for Category, Condition (status), and Lent Status
- Added sorting by Name or Year of Purchase (ascending/descending)
- Implemented getFilteredAndSortedInventory() function for combined filtering and sorting
- Updated inventory count display to show filtered/total items
- UI design matches existing component styling

Mayank447 commented on src/components/admin/InventoryManager.tsx:
```diff
@@ -279,47 +323,136 @@ export default function InventoryManager({
                 <h2 className="text-lg sm:text-xl lg:text-2xl font-bold mb-4 sm:mb-6 text-white uppercase flex items-center gap-2">
                     <CalendarDays size={18} className="sm:w-6 sm:h-6" />
-                    INVENTORY LIST ({inventory.length})
+                    INVENTORY LIST ({getFilteredAndSortedInventory().length} of {inventory.length})
```
When filtering our total also decreases. So use 

```suggestion
                    INVENTORY LIST ({inventory.length})
```

Mayank447 commented on src/components/admin/InventoryManager.tsx:
```diff
@@ -279,47 +323,136 @@ export default function InventoryManager({
+                            <option value="all">ALL ITEMS</option>
+                            <option value="lent">LENT ITEMS</option>
+                            <option value="available">AVAILABLE ITEMS</option>
+                        </select>
```
Refer to the filter and sort code for Gallery/Events/Blogs etc. The front end design needs to be the same, so you can reuse that code.

Mayank447 reviewed (changes requested):
Thanks for this PR! Some minor changes needed.

AKSHAY-RSOL committed:
fix: improve inventory filter/sort implementation

- Rename filterStatus to filterCondition for clarity
- Use useMemo to optimize filtering and sorting (avoid redundant computations)
- Fix inventory count display to show total items
- Add improved empty state messages for filtered results
- Ensure consistent styling and user feedback
